backend/src/schema/resolvers/searches.js

```diff
@@ -3,90 +3,200 @@ import { queryString } from './searches/queryString'
 
 // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
 
+const cypherTemplate = (setup) => `
+  CALL db.index.fulltext.queryNodes('${setup.fulltextIndex}', $query)
+  YIELD node AS resource, score
+  ${setup.match}
+  ${setup.whereClause}
+  ${setup.withClause}
+  RETURN 
+  ${setup.returnClause}
+  AS result
+  SKIP $skip
+  ${setup.limit}
+`
+
+const simpleWhereClause =
+  'WHERE score >= 0.0 AND NOT (resource.deleted = true OR resource.disabled = true)'
+
+const postWhereClause = `WHERE score >= 0.0
+  AND NOT (
+    author.deleted = true OR author.disabled = true
+    OR resource.deleted = true OR resource.disabled = true
+    OR (:User {id: $userId})-[:MUTED]->(author)
+  )`
+
+const searchPostsSetup = {
+  fulltextIndex: 'post_fulltext_search',
+  match: 'MATCH (resource:Post)<-[:WROTE]-(author:User)',
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2020-04-16T08:17:46Z" title="Thursday, April 16th 2020, 10:17:46 am +02:00">Apr 16, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -3,90 +3,200 @@ import { queryString } from './searches/queryString'
 
 // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
 
+const cypherTemplate = (setup) => `
+  CALL db.index.fulltext.queryNodes('${setup.fulltextIndex}', $query)
+  YIELD node AS resource, score
+  ${setup.match}
+  ${setup.whereClause}
+  ${setup.withClause}
+  RETURN 
+  ${setup.returnClause}
+  AS result
+  SKIP $skip
+  ${setup.limit}
+`
+
+const simpleWhereClause =
+  'WHERE score >= 0.0 AND NOT (resource.deleted = true OR resource.disabled = true)'
+
+const postWhereClause = `WHERE score >= 0.0
+  AND NOT (
+    author.deleted = true OR author.disabled = true
+    OR resource.deleted = true OR resource.disabled = true
+    OR (:User {id: $userId})-[:MUTED]->(author)
+  )`
+
+const searchPostsSetup = {
+  fulltextIndex: 'post_fulltext_search',
+  match: 'MATCH (resource)<-[:WROTE]-(author:User)',
```


I guess the type is already defined by `fulltextIndex: 'post_fulltext_search' `.

I would prefer to have the type on the `resource ` for clearness and better readability:
`match: 'MATCH (resource:Post)<-[:WROTE]-(author:User)‘,`